### PR TITLE
[command-synthetics] add datadogSite and subdomain to CLI option

### PR DESCRIPTION
**Pull request by KevinYeramian**
### What and why?

- Add `datadogSite` and `subdomain` to CLI option

### CODE REVIEW

File: src/commands/synthetics/__tests__/run-test.test.ts [hunk 1]
```diff
@@ -324,19 +324,23 @@ describe('run-test', () => {
         apiKey: 'fake_api_key',
         appKey: 'fake_app_key',
         configPath: 'fake-datadog-ci.json',
+        datadogSite: 'datad0g.com',
```
Comment (m-rousse) [inline src/commands/synthetics/__tests__/run-test.test.ts:327]:
```suggestion
        datadogSite: 'datadoghq.com',
```

File: src/commands/synthetics/__tests__/run-test.test.ts [hunk 2]
```diff
@@ -345,8 +349,10 @@ describe('run-test', () => {
...
```
Comment (m-rousse) [inline src/commands/synthetics/__tests__/run-test.test.ts:352]:
```suggestion
        datadogSite: 'datadoghq.com',
```

**Review by m-rousse — APPROVED** [review]

File: src/commands/synthetics/__tests__/run-test.test.ts (hunk 1 above)
Comment (KevinYeramian) [reply src/commands/synthetics/__tests__/run-test.test.ts:327]: `datadoghq.com` is the default value in `DEFAULT_COMMAND_CONFIG`. I wanted to test that we correctly override this value

Comment (m-rousse) [reply src/commands/synthetics/__tests__/run-test.test.ts:327]: Then could you use the `.eu` one? 🙏

**Pull request merged**